src/charm.py

```diff
@@ -413,19 +416,52 @@ def __init__(self, *args) -> None:  # type: ignore[no-untyped-def]
             args: Variable list of positional arguments passed to the parent constructor.
         """
         super().__init__(*args)
-        self.framework.observe(self.on.config_changed, self._on_config_changed)
+        self.framework.observe(self.on.config_changed, self._on_config_changed_with_warning)
         self.framework.observe(self.on.describe_ingresses_action, self._describe_ingresses_action)
 
         # 'ingress' relation handling.
         self.ingress = IngressProvides(self)
         # When the 'ingress' is ready to configure, do so.
-        self.framework.observe(self.on.ingress_available, self._on_config_changed)
+        self.framework.observe(self.on.ingress_available, self._on_config_changed_with_warning)
         self.framework.observe(self.on.ingress_broken, self._on_ingress_broken)
 
+        provide_nginx_route(
+            charm=self,
+            on_nginx_route_available=self._on_config_changed,
+            on_nginx_route_broken=self._on_ingress_broken,
+        )
+
+    def _on_config_changed_with_warning(self, event: Any) -> None:
+        """Handle the ingress relation available event.
+
+        The same functionality as _on_config_changed, but will add warning message if there are
+        applications connected via ingress relation.
+
+        Args:
+            event: not used.
+        """
+        self._on_config_changed(event)
+        status = self.unit.status
+        connected_apps = set()
+        for relation in self.model.relations["ingress"]:
+            remote_app = relation.app
+            if remote_app is not None:
+                connected_apps.add(remote_app.name)
+        if connected_apps:
```

Comment:
Would be possible to have an ingress and a nginx-route relation at the same time?

Reply:
From juju's perspective, it can have two relation at the same time. But if two relation provides the same information (service-name ,etc), it might cause errors in the Nginx ingress integrator charm.

Reply:
Do you think is worth checking in this case?

Reply:
The case where an application has both ingress and nginx-route relations is no different than the case of having two ingress relations or two nginx-route relations. Having multiple relations is not an issue. The problem arises when two or more relations provide the same data, such as `service-hostname` or `service-name`. There are already checks in place to prevent duplicate data, so I don't think additional checks are necessary.